sub-packages/bionemo-scdl/src/bionemo/scdl/io/single_cell_collection.py

```diff
@@ -213,24 +213,19 @@ def shape(self) -> Tuple[int, List[int]]:
         return self.number_of_rows(), self.number_of_variables()
 
     def flatten(
-        self,
-        output_path: str,
-        destroy_on_copy: bool = False,
+        self, output_path: str, destroy_on_copy: bool = False, extend_copy_size: int = 10 * 1_024 * 1_024
     ) -> None:
         """Flattens the collection into a single SingleCellMemMapDataset.
 
         Args:
             output_path: location to store new dataset
             destroy_on_copy: Whether to remove the current data_path
-        """
-        output = SingleCellMemMapDataset(
-            output_path,
-            num_elements=self.number_of_rows(),
-            num_rows=self.number_nonzero_values(),
-            mode=Mode.CREATE_APPEND,
-        )
+            extend_copy_size: how much to copy in memory at once
 
-        output.concat(list(self.fname_to_mmap.values()))
+        """
+        output = next(iter(self.fname_to_mmap.values()))
+        single_cell_list = list(self.fname_to_mmap.values())[1:]
+        output.concat(single_cell_list, extend_copy_size=extend_copy_size, output_path=output_path)
```

Comment:
I think this fails if `fname_to_mmap` only has one value, first on the indexing failure, and then on the concat call (`fail_on_empty_index=True` by default), but I think we want the identity if only one value is present.